connectors/webhook/src/main/java/io/camunda/connector/inbound/authorization/JWTChecker.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.inbound.authorization;
+
+import com.auth0.jwk.Jwk;
+import com.auth0.jwk.JwkException;
+import com.auth0.jwk.JwkProvider;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.auth0.jwt.interfaces.JWTVerifier;
+import com.auth0.jwt.interfaces.Verification;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.connector.api.inbound.webhook.WebhookProcessingPayload;
+import io.camunda.connector.inbound.model.WebhookConnectorProperties;
+import io.camunda.connector.runtime.core.feel.FeelEngineWrapper;
+import java.security.interfaces.RSAPublicKey;
+import java.util.Base64;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JWTChecker {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JWTChecker.class);
+
+  public static boolean verify(
+      final WebhookProcessingPayload payload,
+      final WebhookConnectorProperties webhookConnectorProperties,
+      final JwkProvider jwkProvider,
+      final ObjectMapper objectMapper)
+      throws JsonProcessingException {
+    final String jwtToken = JWTChecker.extractJWTFomHeader(payload);
+    DecodedJWT verifiedJWT = JWTChecker.verifyJWT(jwtToken, jwkProvider);
+
+    // parse this as json and expose this FEEL to extract roles, and test that against the
+    // expression
+    byte[] decodedBytes = Base64.getDecoder().decode(verifiedJWT.getPayload());
+    String decodedPayload = new String(decodedBytes);
+    JsonNode jsonNode = objectMapper.readTree(decodedPayload);
+    // FeelParserWrapper.parseIfIsFeelExpressionOrGetOriginal(jsonNode);
+    FeelEngineWrapper feelEngine = new FeelEngineWrapper();
```

Comment:
Do we have to create the `FeelEngineWrapper` everytime we verify a token?

Reply:
This has been changed based on our discussion on Friday